### ISSUE TITLE
Unhandled promise rejection on PS4 WebMAF

### ISSUE THREAD

**Issue by spetrey** [lightning-js/solid-demo-app#12]
I tried running the demo app on a PS4 WebMAF webview. Looks like the polyfill is throwing an error during a fetch? Here's PS4 WebMAF userAgent: 

```
Mozilla/5.0 (PlayStation 4 WebMAF) AppleWebKit/601.2 (KHTML, like Gecko) WebMAF/v3.1.5-0-g6ddb9f4f SDK: (0x10008011u)
```

Here's the console errors that were thrown. All that rendered on the screen was a black screen:
```
Unexpected token '...'. Expected a parameter pattern or a ')' in parameter list.
Unhandled promise rejection
```

<img width="2560" alt="Screenshot 2024-04-11 at 11 34 47 AM" src="https://github.com/lightning-js/solid-demo-app/assets/6411509/e5771cf4-8da5-4f34-aed5-bae52675d5a6">

**Comment by chiefcll** [comment]
Hey - The renderer doesnt support PS4 at the moment @wouterlucas for awareness.

**Comment by wouterlucas** [comment]
correct! PS4 doesn't support WebGL and requires Canvas2D. However I'd expect an error on grabbing the WebGL context from the canvas, this seems like an issue with the polyfills and/or later ES6+ features not being supported. 

What ES target are you building for?

**Comment by chiefcll** [comment]
Solid is setup for Chrome 49 (ES5) - Not sure what PS4 browser supports.

**Comment by spetrey** [comment]
so, as of WebMAF 3.1.5 it appears it the WebMAF webkit environment is ES6 compatible:

> Specifications of the WebKit browser used in the framework for the purpose of creating HTML content is as follows:
> - HTML 5
> - CSS 3
> - ECMAScript 6 (ES6)
> - XML/XSLT supported
> - JPEG, GIF, animation GIF, PNG, transparent PNG, BMP and SVG supported

I admit, there has been a more recent updates to Sony's WebMAF. [Notably, v3.2.0 which modernizes the browser vendor substantially:](https://ps4.siedev.net/projects/webmaf/allreleases)

> WebMAF SDK 3.2.0 (PS4)
> Native SDK information
> - PS4 SDK 10.5
> WebMAF
> - Major update to modernise WebKit to v613.1.14.2.3 (2022)
>  This update brings 6 years of WebKit updates compared to WebMAF v3.1.7.
>  See https://github.com/WebKit/WebKit/tree/safari-613.1.14.2-branch

I'll upgrade to 3.2.0 next week when I have access to a Sony Dev Kit and re-test against the demo again and report back

**Comment by chiefcll** [comment]
OK let me know how it goes. FYI there are two bundles - you're clearly downloading the legacy which should mean the browser doesnt support ES6 import (Chrome 63, Safari 11.1). Non legacy bundle is import support and above.

Also the vite build can be tweaked to ensure safari is working as expected - I've spent the time to get Chrome 49 working. But again my understanding is PS4 will need canvas2d which Renderer doesn't support. So even if you get solid running on there you still won't "see" anything without the renderer supporting PS4 (canvas2d). Renderer requires WebGL which PS4 has disabled.

**Comment by chiefcll** [comment]
PS canvas2d in the renderer is underway... https://github.com/lightning-js/renderer/pull/246

**Comment by spetrey** [comment]
Apologies for not following up here sooner. I tried to take it for a spin on the latest WebMAF but it appears that attempting to sideload this URL (`https://lightning-js.github.io/solid-demo-app/`) on a PS4 results in this `connection failure` screen. Reproducible on Safari Version 17.5 (19618.2.12.11.6): 

<img width="1802" alt="Screenshot 2024-07-08 at 2 46 48 PM" src="https://github.com/lightning-js/solid-demo-app/assets/6411509/37371cb1-f75d-4837-b33b-f5cd35df30b5">

**Comment by spetrey** [comment]
Update: I may have been rate-limited or something. I now see this in the console:

```
[Error] Unhandled Promise Rejection: Error: Unable to create WebGL context
	(anonymous function) (index-BRrVT1Ra.js:1:160134)
	asyncFunctionResume
	(anonymous function)
	promiseReactionJobWithoutPromise
	promiseReactionJob
```

<img width="1280" alt="Screenshot 2024-07-08 at 2 55 23 PM" src="https://github.com/lightning-js/solid-demo-app/assets/6411509/633f54db-a80f-4c2e-89df-d5661c289af0">
